crates/starknet-types-core/src/regular_contract_address.rs

```diff
@@ -0,0 +1,207 @@
+use core::str::FromStr;
+
+use crate::{
+    contract_address::ContractAddress,
+    felt::Felt,
+    patricia_key::{
+        PatriciaKey, PatriciaKeyFromFeltError, PatriciaKeyFromStrError, PATRICIA_KEY_UPPER_BOUND,
+    },
+};
+
+#[repr(transparent)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[cfg_attr(
+    feature = "parity-scale-codec",
+    derive(parity_scale_codec::Encode, parity_scale_codec::Decode)
+)]
+pub struct RegularContractAddress(ContractAddress);
```

Comment:
what is this ?

Reply:
Addresses that are not 0x0 and 0x1. Most users facing application code should never interact with the special addresses. If they do, it is because of a bug. 
It can be enforced at the type level this way.

Reply:
sounds super confusing, why not add just a regular ContractAddress and have specific functions to create 0x0 etc etc ?

Reply:
@0xLucqs the use case I envision is more the opposite of what you are describing.
I'm writing rust code that allows users to interact with some contracts deployed on chain.
By using `RegularContractAddress` instead of `ContractAddress`, I can guarantee that a bug where I send money to `0x0` or `0x1` due to some badly initialized value, or corrupted json deserialization, will never happen.


> why not add just a regular ContractAddress and have specific functions to create 0x1 

This is a good quality of like improvment (hardcoded const for reserved addresses). I will add it